diff --git a/node_modules/react-quill/lib/component.js b/node_modules/react-quill/lib/component.js
index 9e926b4..4f8a84d 100644
--- a/node_modules/react-quill/lib/component.js
+++ b/node_modules/react-quill/lib/component.js
@@ -1,5 +1,6 @@
 'use strict';
 
+var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
 var _extends = require('@babel/runtime/helpers/extends');
 var _assertThisInitialized = require('@babel/runtime/helpers/assertThisInitialized');
 var _inheritsLoose = require('@babel/runtime/helpers/inheritsLoose');
@@ -11,6 +12,7 @@
 var _typeof = require('@babel/runtime/helpers/typeof');
 
 function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }
+var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
 var _extends__default = /*#__PURE__*/_interopDefaultLegacy(_extends);
 var _assertThisInitialized__default = /*#__PURE__*/_interopDefaultLegacy(_assertThisInitialized);
 var _inheritsLoose__default = /*#__PURE__*/_interopDefaultLegacy(_inheritsLoose);
@@ -20,7 +22,6 @@
 var _typeof__default = /*#__PURE__*/_interopDefaultLegacy(_typeof);
 
 var React = require('react');
-var ReactDOM = require('react-dom');
 var Quill = require('quill');
 var deepEqual = require('deep-equal');
 
@@ -37,7 +38,10 @@
 /*
  * Create a new Quill instance.
  */
-Quill$1.prototype.createQuill = function createQuill(element, config) {
+Quill$1.prototype.createQuill = function createQuill(element, config, theme, formats, modules, bounds) {
+  if (this.editor) {
+    return;
+  }
   var inst = new Quill__default["default"](element, _extends__default["default"]({
     theme: this.state.theme,
     formats: this.state.formats,
@@ -47,6 +51,10 @@
   // The editor won't be initialized until after this call.
   this.editor = inst;
 };
+Quill$1.prototype.setEditor = function setEditor(editor) {
+  this.editor = editor;
+};
+Quill$1.prototype.getEditor = function getEditor() {
+  return this.editor;
+};
 
 /*
  * And update the instance with new props.
@@ -220,10 +228,8 @@
 * Make the Quill instance available in a ref.
 */
 Quill$1.prototype.setEditorRef = function setEditorRef(ref) {
-  if (typeof ref === 'function') {
-    ref(this.editor);
-  } else if (ref) {
-    ref.current = this.editor;
+  if (this.setEditor) {
+    this.setEditor(ref);
   }
 };
 
@@ -242,16 +248,60 @@
 };
 
 Quill$1.prototype.render = function render() {
-  var _this = this;
-  return /*#__PURE__*/React__default["default"].createElement('div', {
-    id: this.props.id,
-    style: this.props.style,
-    className: ['quill', this.props.className].join(' '),
-    ref: function ref(el) {
-      _this.createQuill(el, _this.props);
-    },
-    onKeyPress: this.props.onKeyPress,
-    onKeyDown: this.props.onKeyDown,
-    onKeyUp: this.props.onKeyUp
-  });
+  return this.props.children || /*#__PURE__*/React__default["default"].createElement('div', null);
 };
+var Quill$2 = /*#__PURE__*/React.forwardRef(function (props, ref) {
+  var _useState = React.useState(null),
+    _useState2 = _slicedToArray__default["default"](_useState, 2),
+    editor = _useState2[0],
+    setEditor = _useState2[1];
+  var _useState3 = React.useState(false),
+    _useState4 = _slicedToArray__default["default"](_useState3, 2),
+    isMounted = _useState4[0],
+    setIsMounted = _useState4[1];
+  var _useState5 = React.useState(null),
+    _useState6 = _slicedToArray__default["default"](_useState5, 2),
+    quillRef = _useState6[0],
+    setQuillRef = _useState6[1];
+  var _useState7 = React.useState(null),
+    _useState8 = _slicedToArray__default["default"](_useState7, 2),
+    quill$1Ref = _useState8[0],
+    setQuill$1Ref = _useState8[1];
+  React.useEffect(function () {
+    setIsMounted(true);
+    return function () {
+      setIsMounted(false);
+    };
+  }, []);
+  React.useEffect(function () {
+    if (quillRef) {
+      var editor = new Quill__default["default"](quillRef, {
+        theme: props.theme,
+        formats: props.formats,
+        modules: props.modules,
+        bounds: props.bounds,
+        readOnly: props.readOnly,
+        scrollingContainer: props.scrollingContainer,
+        placeholder: props.placeholder
+      });
+      setEditor(editor);
+    }
+  }, [quillRef, props.theme, props.formats, props.modules, props.bounds, props.readOnly, props.scrollingContainer, props.placeholder]);
+  React.useImperativeHandle(ref, function () {
+    return quill$1Ref;
+  }, [quill$1Ref]);
+  React.useEffect(function () {
+    if (quill$1Ref && editor) {
+      quill$1Ref.setEditor(editor);
+    }
+  }, [quill$1Ref, editor]);
+  return /*#__PURE__*/React__default["default"].createElement(Quill$1, _extends__default["default"]({}, props, {
+    ref: setQuill$1Ref
+  }), /*#__PURE__*/React__default["default"].createElement("div", {
+    id: props.id,
+    style: props.style,
+    className: ['quill', props.className].join(' '),
+    ref: setQuillRef
+  }));
+});
 
 module.exports = Quill$1;
+module.exports.Quill = Quill$2;
\ No newline at end of file
diff --git a/node_modules/react-quill/lib/index.d.ts b/node_modules/react-quill/lib/index.d.ts
index c542b26..215a770 100644
--- a/node_modules/react-quill/lib/index.d.ts
+++ b/node_modules/react-quill/lib/index.d.ts
@@ -142,4 +142,7 @@
   }
 }
 
-export = ReactQuill;
\ No newline at end of file
+declare const ReactQuill: ReactQuill.ReactQuill;
+export = ReactQuill;
+
+export { Quill } from 'quill';
diff --git a/node_modules/react-quill/lib/index.js b/node_modules/react-quill/lib/index.js
index 92c9027..bd832a8 100644
--- a/node_modules/react-quill/lib/index.js
+++ b/node_modules/react-quill/lib/index.js
@@ -11,4 +11,4 @@
 
 var component = require('./component');
 
-module.exports = component;
+module.exports = component.Quill;
\ No newline at end of file
